### ISSUE TITLE
关于swift微调 Seed-OSS-36B-Instruct 思考预算的问题

### ISSUE THREAD

**Issue by wa3926** [modelscope/ms-swift#6193]
使用 swift微调 Seed-OSS-36B-Instruct 然后使用swift export 合并  经测试发现思考响应内容没了

<img width="1106" height="478" alt="Image" src="https://github.com/user-attachments/assets/2f5af06f-29bd-43a3-aff8-699e1f9eebaa" />

以下是微调代码 
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '0,1,2,3'

from swift.llm import get_model_tokenizer, load_dataset, get_template, EncodePreprocessor
from swift.utils import get_logger, find_all_linears, get_model_parameter_info, plot_images, seed_everything
from swift.tuners import Swift, LoraConfig
from swift.trainers import Seq2SeqTrainer, Seq2SeqTrainingArguments
from functools import partial

logger = get_logger()
seed_everything(42)

# 模型
model_id_or_path = '/ai/modelscope/models/ByteDance-Seed/Seed-OSS-36B-Instruct'  # model_id or model_path
system = 'You are a helpful assistant.'
output_dir = '/ai/wjb/output'

# 数据集
dataset = ['AI-ModelScope/alpaca-gpt4-data-zh#500', 'AI-ModelScope/alpaca-gpt4-data-en#500',
           'swift/self-cognition#500']  # dataset_id or d

# ataset_path
data_seed = 42
max_length = 2048
split_dataset_ratio = 0.01  # 切分验证集
num_proc = 4  # 预处理的进程数
# 替换自我认知数据集中的填充符：{{NAME}}, {{AUTHOR}}
model_name = ['小淮', 'Xiao Huai']  # 模型的中文名和英文名
model_author = ['楚淮科技', 'Chkj']  # 模型作者的中文名和英文名

# lora
lora_rank = 8
lora_alpha = 32
#  dataloader_num_workers=1	数据加载的进程数（多进程加速数据读取，1 表示单进程，适合调试或数据量小的场景）。
# 训练超参数
training_args = Seq2SeqTrainingArguments(
    output_dir=output_dir,
    learning_rate=1e-4,
    per_device_train_batch_size=4,
    per_device_eval_batch_size=4,
    gradient_checkpointing=True,
    weight_decay=0.1,
    lr_scheduler_type='cosine',
    warmup_ratio=0.05,
    report_to=['tensorboard'],
    logging_first_step=True,
    save_strategy='steps',
    save_steps=50,
    eval_strategy='steps',
    eval_steps=50,
    gradient_accumulation_steps=1,
    num_train_epochs=1,
    metric_for_best_model='loss',
    save_total_limit=2,
    logging_steps=5,
    dataloader_num_workers=8,
    data_seed=data_seed,
)

#output_dir = os.path.abspath(os.path.expanduser(output_dir))
logger.info(f'output_dir: {output_dir}')

# 获取模型和对话template，并将可训练的lora层加入到模型中：
model, tokenizer = get_model_tokenizer(model_id_or_path)
logger.info(f'model_info: {model.model_info}')
template = get_template(model.model_meta.template, tokenizer, default_system=system, max_length=max_length)
template.set_mode('train')

target_modules = find_all_linears(model)
# 因果语言建模 模型根据前面的 token 来预测下一个 token
lora_config = LoraConfig(task_type='CAUSAL_LM', r=lora_rank, lora_alpha=lora_alpha,
                         target_modules=target_modules)
model = Swift.prepare_model(model, lora_config)
logger.info(f'lora_config: {lora_config}')

# 打印模型结构和训练的参数量
logger.info(f'model: {model}')
model_parameter_info = get_model_parameter_info(model)
logger.info(f'model_parameter_info: {model_parameter_info}')

# 下载并载入数据集，并切分成训练集和验证集，然后将文本编码成tokens：
train_dataset, val_dataset = load_dataset(dataset, split_dataset_ratio=split_dataset_ratio, num_proc=num_proc,
        model_name=model_name, model_author=model_author, seed=data_seed)

logger.info(f'train_dataset: {train_dataset}')
logger.info(f'val_dataset: {val_dataset}')
logger.info(f'train_dataset[0]: {train_dataset[0]}')

train_dataset = EncodePreprocessor(template=template)(train_dataset, num_proc=num_proc)
val_dataset = EncodePreprocessor(template=template)(val_dataset, num_proc=num_proc)
logger.info(f'encoded_train_dataset[0]: {train_dataset[0]}')

# 打印一条样本
template.print_inputs(train_dataset[0])

#初始化trainer并开始训练：
model.enable_input_require_grads()  # 兼容gradient checkpointing
trainer = Seq2SeqTrainer(
    model=model,
    args=training_args,
    data_collator=template.data_collator,
    train_dataset=train_dataset,
    eval_dataset=val_dataset,
    template=template,
)
trainer.train()
last_model_checkpoint = trainer.state.last_model_checkpoint
logger.info(f'last_model_checkpoint: {last_model_checkpoint}')

**Comment by tastelikefeet** [comment]
有几个建议：
1. 这几个数据集都不包含思考过程，请使用一个包含了<think></think>的数据集试一试
2. 测试一下，如果不合并是否也没有思考能力，排除下是export带来的问题还是sft带来的问题

**Comment by wa3926** [comment]
@tastelikefeet  加了 ignore_empty_think 好像好了  没合并前也有这个问题   暂时没有思考的数据集  本地准备的基本都是非思考的数据集   谢谢！
